src/main/java/com/fasterxml/jackson/databind/util/RecordUtil.java

```diff
@@ -0,0 +1,81 @@
+package com.fasterxml.jackson.databind.util;
+
+import com.fasterxml.jackson.databind.introspect.AnnotatedClass;
+import com.fasterxml.jackson.databind.introspect.AnnotatedConstructor;
+
+import java.lang.reflect.Method;
+import java.util.Arrays;
+
+/**
+ * Helper class to detect Java records without Java 14 as Jackson targets is Java 8.
+ * <p>
+ * See <a href="https://openjdk.java.net/jeps/359">JEP 359</a>
+ */
+public final class RecordUtil {
+
+    private static final String RECORD_CLASS_NAME = "java.lang.Record";
+    private static final String RECORD_GET_RECORD_COMPONENTS = "getRecordComponents";
+
+    private static final String RECORD_COMPONENT_CLASS_NAME = "java.lang.reflect.RecordComponent";
+    private static final String RECORD_COMPONENT_GET_NAME = "getName";
+    private static final String RECORD_COMPONENT_GET_TYPE = "getType";
+
+    public static boolean isRecord(Class<?> aClass) {
+        return aClass != null
+                && aClass.getSuperclass() != null
+                && aClass.getSuperclass().getName().equals(RECORD_CLASS_NAME);
+    }
+
+    /**
+     * @return Record component's names, ordering is preserved.
+     */
+    public static String[] getRecordComponents(Class<?> aRecord) {
+        if (!isRecord(aRecord)) {
+            return new String[0];
+        }
+
+        try {
+            Method method = Class.class.getMethod(RECORD_GET_RECORD_COMPONENTS);
```

Comment:
Should be cached so that this lookup is not executed for every access, but I guess that can be done as a post-cleanup.